distributed/worker.py

```diff
@@ -1440,15 +1439,12 @@ async def start_unsafe(self):
         self.start_periodic_callbacks()
         return self
 
-    async def terminate(self, **kwargs):
-        return await self.close(nanny=True, **kwargs)
-
```

Comment:
I opted to remove `Worker.terminate` since it won't be used now that the `Worker.close` default is to also close the Nanny. I think we will want two different methods eventually, but found the naming of `terminate` vs `close` very confusing—it should be something like `terminate` vs `maybe_restart`?

If others would prefer to see `terminate` stick around, I'm happy to remove this commit and leave it.